sources/packages/api/src/services/student-appeal/student-appeal.service.ts

```diff
@@ -128,4 +139,162 @@ export class StudentAppealService extends RecordDataModelService<StudentAppeal>
+      .where("studentAppeal.id = :appealId", { appealId })
+      .getRawAndEntities();
+
+    const [appealWithStatus] = mapFromRawAndEntities<StudentAppealWithStatus>(
+      queryResult,
+      "status",
+    );
+    return appealWithStatus;
+  }
+
+  /**
+   * Builds the conditions to define the status of a student appeal
+   * based on the statuses of its appeal requests.
+   * @returns SQL select to be used in a query to return the appeal status.
+   */
+  private buildStatusSelect(): string {
+    return `CASE
+      WHEN EXISTS(${this.studentAppealRequestsService
+        .appealsByStatusQueryObject(StudentAppealStatus.Pending)
+        .getSql()}) THEN '${StudentAppealStatus.Pending}'
+      WHEN EXISTS(${this.studentAppealRequestsService
+        .appealsByStatusQueryObject(StudentAppealStatus.Approved)
+        .getSql()}) THEN '${StudentAppealStatus.Approved}'
+      ELSE '${StudentAppealStatus.Declined}'
+    END`;
+  }
+
+  /**
+   * Update all student appeals requests at once.
+   * @param appealId appeal if to be retrieved.
+   * @param approvals all appeal requests that must be updated with
+   * an approved/declined status. All requests that belongs to the
+   * appeal must be provided.
+   * @param auditUserId user that should be considered the one that is
+   * causing the changes.
+   */
+  async approveRequests(
+    appealId: number,
+    approvals: StudentAppealRequestApproval[],
+    auditUserId: number,
+  ): Promise<StudentAppeal> {
+    const appealToUpdate = await this.repo
+      .createQueryBuilder("studentAppeal")
+      .select([
+        "studentAppeal.id",
+        "appealRequest.id",
+        "appealRequest.appealStatus",
+        "studentAssessment.id",
+        "application.id",
+        "application.data",
+      ])
+      .innerJoin("studentAppeal.appealRequests", "appealRequest")
+      .innerJoin("studentAppeal.application", "application")
+      .leftJoin("studentAppeal.studentAssessment", "studentAssessment")
+      .where("studentAppeal.id = :appealId", { appealId })
+      .getOne();
+
+    if (!appealToUpdate) {
+      throw new CustomNamedError(
+        "Not able to find the appeal.",
+        STUDENT_APPEAL_NOT_FOUND,
+      );
+    }
+
+    if (appealToUpdate.studentAssessment) {
+      throw new CustomNamedError(
+        "An assessment was already created to this student appeal.",
+        STUDENT_APPEAL_INVALID_OPERATION,
+      );
+    }
+
+    const auditUser = { id: auditUserId } as User;
+    const auditDate = new Date();
+
+    // If a students appel has, for instance, 3 requests, all must be updated at once.
+    // This counter checks if all the requests were updated as expected.
+    let updatedRecords = 0;
+    appealToUpdate.appealRequests.forEach((appealRequest) => {
+      // Verify if the appeal request to be updated is in the right state.
+      if (appealRequest.appealStatus !== StudentAppealStatus.Pending) {
+        throw new CustomNamedError(
```

Comment:
I see a list of validations happening here. 
   1.Check pending status

    2.Iterating to find the appealRequestPayload to update by matching id

    3.Validating if all the appeal requests are updated at once.

IMO, if all these validations can be delegated to one single query, then we can just form an array of objects as appealRequest[] and to repo.save.

Reply:
I will try to move the validations to a query. Just to be clear these are the variations happening:
1 - Ensure that ALL requests on DB are on pending status;
2 - Ensure that the API payload has all requests to be updated;
3 - Ensure that API did not receive more requests than the ones supposed to be updated;
4 - Ensure that all are updated at once.
Another point is that moving the validations to the DB, in this case, will make the error messages more generic and keeping in mind that get all the requests, in this case, are about, many times, retrieving one record from DB.
Either way, I will give it a try 😉

Reply:
Method refactored and some validations moved to the query.